### PR TITLE
NumPy: rework, bump, cleanup, extra libs

### PR DESCRIPTION
**Pull request by extrowerk**
Refer: https://discuss.haiku-os.org/t/package-python3-numpy-not-working-on-nightly-build-hrev52992/8380

Import works now, but both version (python2 and 3) fails with missing symbols:
* Python2: cblas_call_sub
* Python3: cblas_sgemm

Both symbol defined in `libopenblas-r0.3.5.so`:
```
/s/lib » objdump -x libopenblasp-r0.3.5.so| ag cblas_sgemm
00000000000b07e0 g     F .text  00000000000006d1              cblas_sgemm
/s/lib » objdump -x libopenblasp-r0.3.5.so| ag cblas_cdotc_sub
00000000000c13b0 g     F .text  00000000000000b2              cblas_cdotc_sub
```
Disabling blas results Python crash at import.

### CODE REVIEW

**Comment by extrowerk** [comment]
@Begasus can you pls check blis on x86?

**Comment by hanya** [comment]
How did you obtain the missing symbol error? I tried to run ```numpy.test()``` but no such error happened.
But I needed to patch to avoid invalid opcode (xgetbv, GCC 7.3.0 problem) while import the module.

**Comment by extrowerk** [comment]
I'll try to add your patch and retest it.

**Comment by extrowerk** [comment]
Now it works ok, at least no problem at importing it.
I would say, it would be better to merge mine, as it is uniformized, one recipe for both python version, but i'm open for other ideas.

**Comment by extrowerk** [comment]
User reports: working.
https://discuss.haiku-os.org/t/package-python3-numpy-not-working-on-nightly-build-hrev52992/8380/6?u=extrowerk

File: dev-python/numpy/numpy-1.16.2.recipe [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"
+
+PYTHON_PACKAGES=(python python3)
+PYTHON_VERSIONS=(2.7 3.6)
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	${portName}_$pythonPackage = $portVersion\n\
+	cmd:f2py\n\
```
Comment (alaviss) [inline dev-python/numpy/numpy-1.16.2.recipe]: Wouldn't this mean that the two python versions will conflict?

File: dev-python/numpy/numpy-1.16.2.recipe [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	python_numpy
+	"
+
+INSTALL()
+{
+# Check at every update in site.cfg.example!
+cat > site.cfg << EOF
+[ALL]
+libraries = execinfo,lapack,openblas
+library_dirs = /system/`echo ${relativeDevelopLibDir}`
```
Comment (alaviss) [inline dev-python/numpy/numpy-1.16.2.recipe]:
```suggestion
library_dirs = /system/$relativeDevelopLibDir
```

It'd be shorter to just use the variables directly :P

I'm not a fan of hardcoding, but it seems to be rather complex to patch this up using `findpaths`, so it's fine for this case.

File: dev-python/numpy/numpy-1.16.2.recipe (hunk 1 above)
Comment (extrowerk) [reply dev-python/numpy/numpy-1.16.2.recipe]: True

File: dev-python/numpy/numpy-1.16.2.recipe (hunk 2 above)
Comment (alaviss) [reply dev-python/numpy/numpy-1.16.2.recipe]: Can you do the same with all other variables?

File: dev-python/numpy/numpy-1.16.2.recipe [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+HOMEPAGE="http://numpy.scipy.org/"
+COPYRIGHT="2005-2019 Travis E. Oliphant et al."
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/numpy/numpy/releases/download/v$portVersion/numpy-$portVersion.tar.gz"
+CHECKSUM_SHA256="8088221e6e27da8d5907729f0bfe798f526836f22cc59ae83a0f867e67416a3e"
+SOURCE_DIR="numpy-$portVersion"
+PATCHES="numpy-$portVersion.patch"
+
+ARCHITECTURES="any"
```
Comment (korli) [inline dev-python/numpy/numpy-1.16.2.recipe]: This shouldn't be any. $secondaryArchSuffix will only work with a secondary architecture defined anyway.

File: dev-python/numpy/numpy-1.16.2.recipe [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+PROVIDES="
+	$portName = $portVersion
```
Comment (Begasus) [inline dev-python/numpy/numpy-1.16.2.recipe]: Why use $portName here?

File: dev-python/numpy/numpy-1.16.2.recipe (hunk 3 above)
Comment (Begasus) [reply dev-python/numpy/numpy-1.16.2.recipe]: blis can't be build for x86_gcc2, so agreed, any shouldn't be used here

File: dev-python/numpy/numpy-1.16.2.recipe (hunk 4 above)
Comment (extrowerk) [reply dev-python/numpy/numpy-1.16.2.recipe]: The recipe template contains it.

File: dev-python/numpy/numpy-1.16.2.recipe [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+	lib:libblis$secondaryArchSuffix
+	lib:libexecinfo$secondaryArchSuffix
+#	lib:libfftw3$secondaryArchSuffix
+	lib:liblapack$secondaryArchSuffix
+	lib:libopenblas$secondaryArchSuffix
+#	lib:libumfpack$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku_devel$secondaryArchSuffix
```
Comment (Begasus) [inline dev-python/numpy/numpy-1.16.2.recipe]: Some mixup? :)
```diff
-	haiku_devel$secondaryArchSuffix
+	haiku${secondaryArchSuffix}_devel
```

**Comment by extrowerk** [comment]
Seems to be ok, at least it works, unlike the earlier, so lets merge it.

**Pull request merged**